pkgs/by-name/pa/parabolic/package.nix

```
@@ -1,77 +1,119 @@
-  nugetDeps = ./deps.json;
-  executables = "NickvisionTubeConverter.GNOME";
+  buildInputs =
+    [
+      libnick
+      boost
+    ]
+    ++ (
+      if useQtInterface then
+        [
+          qt6.qtbase
+          qt6.qtsvg
+        ]
+      else
+        [
+          glib
+          gtk4
+          libadwaita
+          libxmlxx5
+        ]
+    );
 
-   nativeBuildInputs = [
-    pkg-config
-    wrapGAppsHook4
-    glib
-    shared-mime-info
-    gdk-pixbuf
-    blueprint-compiler
+  cmakeFlags = [
+    (lib.cmakeFeature "UI_PLATFORM" (if useQtInterface then "qt" else "gnome"))
   ];
 
-  buildInputs = [ gtk4 libadwaita ];
+  preFixup =
+    (if useQtInterface then "qtWrapperArgs" else "gappsWrapperArgs")
+    + ''+=(--prefix PATH : ${
+      lib.makeBinPath [
+        aria2
+        ffmpeg
+        yt-dlp
+      ]
+    })'';
 
-  runtimeDeps = [
-    gtk4
-    libadwaita
-    glib
-    gdk-pixbuf
-  ];
-
-  postPatch = ''
-    substituteInPlace NickvisionTubeConverter.Shared/Linux/org.nickvision.tubeconverter.desktop.in --replace '@EXEC@' "NickvisionTubeConverter.GNOME"
-  '';
-
-  postInstall = ''
-    install -Dm444 NickvisionTubeConverter.Shared/Resources/org.nickvision.tubeconverter.svg -t $out/share/icons/hicolor/scalable/apps/
-    install -Dm444 NickvisionTubeConverter.Shared/Resources/org.nickvision.tubeconverter-symbolic.svg -t $out/share/icons/hicolor/symbolic/apps/
-    install -Dm444 NickvisionTubeConverter.Shared/Linux/org.nickvision.tubeconverter.desktop.in -T $out/share/applications/org.nickvision.tubeconverter.desktop
-  '';
-
-  makeWrapperArgs = [ "--prefix PATH : ${lib.makeBinPath [ pythonEnv ffmpeg ]}" ];
+  passthru.updateScript = nix-update-script { };
 
-  passthru.updateScript = ./update.sh;
+  meta = {
+    description = "Graphical frontend for yt-dlp to download video and audio";
+    longDescription = ''
+      Parabolic is a user-friendly frontend for `yt-dlp` that supports
+      many features including but limited to:
+      - Downloading and converting videos and audio using ffmpeg.
+      - Supporting multiple codecs.
+      - Offering YouTube sponsorblock support.
+      - Running multiple downloads at a time.
+      - Downloading metadata and video subtitles.
+      - Allowing the use of `aria2` for parallel downloads.
+      - Offering a graphical keyring to manage account credentials.
+      - Being available as both a Qt and GNOME application.
 
-  meta = with lib; {
-    description = "Download web video and audio";
+      By default, the GNOME interface is used, but the Qt interface can
+      be built by overriding the `useQtInterface` parameter to `true`.
+    '';
     homepage = "https://github.com/NickvisionApps/Parabolic";
-    license = licenses.mit;
-    maintainers = with maintainers; [ ewuuwe ];
-    mainProgram = "NickvisionTubeConverter.GNOME";
-    platforms = platforms.linux;
+    license = lib.licenses.gpl3Plus;
+    maintainers = [ lib.maintainers.normalcea ];
```

```suggestion
    maintainers = with lib.maintainers; [
      normalcea
      getchoo
    ];
```

Would still love to help out :)